### ISSUE TITLE
.lastChild()

### ISSUE THREAD

**Issue by msklvsk** [libxmljs/libxmljs#420]
There is no `.lastChild()` method and I can't find any constant-time workaraounds (without xpath).
Am I missing something?

**Comment by rchipka** [comment]
You could make your own `.lastChild()`:

```javascript
libxmljs.Element.prototype.lastChild = function() {
    return this.get('./child::*[last()]');
}
```

or

```javascript
libxmljs.Element.prototype.lastChild = function() {
    var children = this.childNodes();
    return children[children.length - 1];
}
```

**Comment by msklvsk** [comment]
But isn't the first example an overkill and the second one still linear time (I guess?). It would fill an array for you, while all you wanted was the last child, a reference which libxml internally holds.

**Comment by rchipka** [comment]
Yes, none of these solutions will ever be as good as a native one.

You could submit a PR adding `.firstChild` and `.lastChild` to the API. I see no reason why it wouldn't be merged in.
